server/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java

```diff
@@ -1134,4 +1135,462 @@ public void test_update_continues_on_error() {
+        execute("""
+            create table t (
+                a int generated always as c+1,
+                b int generated always as round(random() * 10),
+                i int,
+                o object as (
+                    a int generated always as c+1,
+                    b int generated always as round(random() * 10),
+                    i int,
+                    o object as (
+                        a int generated always as c+1,
+                        b int generated always as round(random() * 10),
+                        i int
+                    )
+                ),
+                c int
+            )
+            """);
+        execute("insert into t values (11, 2, 3, {a=11, b=5, i=6, o={a=11, b=8, i=9}}, 10)");
+        execute("refresh table t");
+        execute("select * from t");
+        assertThat(response).hasRows("11| 2| 3| {a=11, b=5, i=6, o={a=11, b=8, i=9}}| 10");
+
+        execute("update t set a=101, b=22, i=33, o={a=101, b=55, i=66, o={a=101, b=88, i=99}}, c=100");
+        execute("refresh table t");
+        execute("select * from t");
+        assertThat(response).hasRows("101| 22| 33| {a=101, b=55, i=66, o={a=101, b=88, i=99}}| 100");
+
+        execute("update t set a=11, b=2, i=3, o={a=11, b=5, i=6, o={a=11, b=8, i=9}}, c=10 returning *");
+        assertThat(response).hasRows("11| 2| 3| {a=11, b=5, i=6, o={a=11, b=8, i=9}}| 10");
+    }
+
+    @Test
+    public void test_can_assign_to_generated_columns_via_subscript_expression() {
+        // any values to nondeterministic generated columns and only valid values to deterministic generated columns
+        execute("""
+            create table t (
+                a int generated always as c+1,
+                b int generated always as round(random() * 10),
+                i int,
+                o object as (
+                    a int generated always as c+1,
+                    b int generated always as round(random() * 10),
+                    i int,
+                    o object as (
+                        a int generated always as c+1,
+                        b int generated always as round(random() * 10),
+                        i int
+                    )
+                ),
+                c int
+            )
+            """);
+        execute("insert into t values (11, 2, 3, {a=11, b=5, i=6, o={a=11, b=8, i=9}}, 10)");
+        execute("refresh table t");
+        execute("select * from t");
+        assertThat(response).hasRows("11| 2| 3| {a=11, b=5, i=6, o={a=11, b=8, i=9}}| 10");
+
+        // nondeterministic generated columns are always re-generated: b, o['b'] or can be assigned: o['o']['b']
+        execute("update t set o['o']['a']=11, o['o']['b']=12, o['o']['i']=13 returning b=-1, o['b']=-1, o['o']");
+        assertThat(response).hasRows("false| false| {a=11, b=12, i=13}");
+    }
+
+    @Test
+    public void test_can_assign_nondeterministic_generated_columns_to_null() {
```

Comment:
The tests overall look very similar - are these all really unique scenarios with no overlap between them?